adafruit_pn532/i2c.py

```diff
@@ -83,7 +83,6 @@ def _wait_ready(self, timeout=1):
                 with self._i2c:
                     self._i2c.readinto(status)
             except OSError:
-                self._wakeup()
```

Comment:
dont remove this?

Reply:
It seems that when you issue a command to the PN532 `_wait_ready` is called before it is ready to respond, which results in an OSError. I tried to check in CP what error was being triggered and it was a NACK. When I removed this line it ends up only taking a few milliseconds to send the ready status. But with this line, it got stuck sleeping for half a second. And if there is no `_req` pin setup, the only thing that `self._wakeup` does is sleep for half a second, which is overkill for here.

It is also worth noting that this problem didn't present itself on older versions of CP for me. I think some of the newer performance improvements are the reason this is surfacing. `_wait_ready` was never called fast enough after a command to cause the OSError to be thrown like it is now. Sooo that's kind of cool lol.

Reply:
> It seems that when you issue a command to the PN532 `_wait_ready` is called before it is ready to respond, which results in an OSError. I tried to check in CP what error was being triggered and it was a NACK. When I removed this line it ends up only taking a few milliseconds to send the ready status. But with this line, it got stuck sleeping for half a second. And if there is no `_req` pin setup, the only thing that `self._wakeup` does is sleep for half a second, which is overkill for here.

Why not return early from `_wakeup` if `_req` isn't available? Seems like we do want the ability to sleep and wake the device.

> It is also worth noting that this problem didn't present itself on older versions of CP for me. I think some of the newer performance improvements are the reason this is surfacing. `_wait_ready` was never called fast enough after a command to cause the OSError to be thrown like it is now. Sooo that's kind of cool lol.

🎉 Does the OSError vary depending on whether it's asleep or waking up?

Reply:
> Why not return early from _wakeup if _req isn't available? Seems like we do want the ability to sleep and wake the device.

That seems like a good idea.

> 🎉 Does the OSError vary depending on whether it's asleep or waking up?

It doesn't seem to matter. I'm not 100% sure, but when running one of the simple examples each time you loop through it you always get the same OSError.

I've been looking through the PN532 manual, and while I definitely am not familiar with reading this kind of thing, so I could be wrong, but I think the issue doesn't have to do with waking up. At least not the problem I was seeing.

<img width="786" alt="Screen Shot 2020-08-25 at 8 51 00 PM" src="https://user-images.githubusercontent.com/11963090/91242275-fee4ef80-e714-11ea-831a-6ce343d13fe4.png">

I think waking up happens in between `I2C address Write` and `COMMAND frame`. The OSError is being thrown when we try to read in between `COMMAND frame` and `I2C address Read + Acknowledge frame`. So wouldn't calling `_wakeup` be unnecessary?

Reply:
I don't know anything about the sensor itself. Does it power down automatically or can we track when the driver shuts it down?